File: src/main/java/org/tindalos/principle/infrastructure/plugin/Checks.java
package org.tindalos.principle.infrastructure.plugin;

import org.tindalos.principle.domain.expectations.DesignQualityExpectations;
import org.tindalos.principle.domain.expectations.Layering;
import org.tindalos.principle.domain.expectations.PackageCoupling;
import org.tindalos.principle.domain.expectations.SubmodulesBlueprint;

public class Checks implements DesignQualityExpectations {
	
	private Layering layering;
	private PackageCoupling packageCoupling;
	private SubmodulesBlueprint submodulesBlueprint;
	
	public Layering getLayering() {
		return layering;
	}
	
	public void setLayering(Layering layering) {
		this.layering = layering;
	}
	
	public PackageCoupling getPackageCoupling() {
		return packageCoupling;
	}
	
	public void setPackageCoupling(PackageCoupling packageCoupling) {
		this.packageCoupling = packageCoupling;
	}

    public SubmodulesBlueprint getSubmodulesBlueprint() {
        return this.submodulesBlueprint;
    }
    
    public void setSubmodulesBlueprint(SubmodulesBlueprint submodulesBlueprint) {
        this.submodulesBlueprint = submodulesBlueprint;
    }
}

File: src/main/java/org/tindalos/principle/infrastructure/plugin/Checks.scala
package org.tindalos.principle.infrastructure.plugin

import org.tindalos.principle.domain.expectations._

class Checks(
              var layering: Layering = null,
              var thirdParty: ThirdParty = null,
              var packageCoupling: PackageCoupling = null,
              var submodulesBlueprint: SubmodulesBlueprint = null)
  extends DesignQualityExpectations {

  def this(aPackageCoupling: PackageCoupling) = this(packageCoupling=aPackageCoupling)

  def this(aLayering: Layering) = this(layering=aLayering)

  def this(aThirdParty: ThirdParty) = this(thirdParty=aThirdParty)

  def this(aLayering: Layering, aThirdParty: ThirdParty) = this(layering=aLayering, thirdParty=aThirdParty)

  def this() = this(null, null, null)

}